src/engine/objects/robot/SimRobotWheel.ts

```diff
@@ -110,4 +110,51 @@ export class SimRobotWheel extends SimObject {
   getFixtureDef(): FixtureDef {
     return this._fixtureSpecs;
   }
+
+  private getForwardVelocity(): Vec2 {
+    const currForwardNormal = this._body.getWorldVector(new Vec2(0, 1));
+    return currForwardNormal.mul(
+      Vec2.dot(currForwardNormal, this._body.getLinearVelocity())
+    );
+  }
+
+  private getLateralVelocity(): Vec2 {
+    const currRightNormal = this._body.getWorldVector(new Vec2(1, 0));
+    return currRightNormal.mul(
+      Vec2.dot(currRightNormal, this._body.getLinearVelocity())
+    );
+  }
+
+  private updateFriction(): void {
+    // Lateral
+    // TODO the max lateral impulse should be proportional to mass
+    const maxLateralImpulse = 2.0;
+    let impulse = this.getLateralVelocity().mul(
+      -this._body.getMass() * this._sideSlipPercent
+    );
+
+    if (impulse.length() > maxLateralImpulse) {
+      impulse = impulse.mul(maxLateralImpulse / impulse.length());
+    }
+    this._body.applyLinearImpulse(impulse, this._body.getWorldCenter());
+
+    // Angular
+    this._body.applyAngularImpulse(
```

Comment:
i'm not sure how much torque would be generated at this point, this is a twisting force on the wheel itself...? i.e. steering the wheel about its center.?